Test failure on Python 3.11

`test_snitun_runner_updown` fails on Python 3.11. At first glances it looks related to those changes (https://docs.python.org/3/library/asyncio-task.html#waiting-primitives).

```bash
(snitun) [fab@python311-tester snitun]$ bin/pytest 
============================================================ test session starts ============================================================
platform linux -- Python 3.11.2, pytest-7.2.2, pluggy-1.0.0
rootdir: /home/fab/Documents/repos/snitun, configfile: setup.cfg
plugins: asyncio-0.20.3, aiohttp-1.0.4
asyncio: mode=Mode.AUTO
collected 104 items                                                                                                                         

tests/client/test_client_peer.py ......                                                                                               [  5%]
[...]
tests/server/test_peer_manager.py .........                                                                                           [ 69%]
tests/server/test_run.py F..........                                                                                                  [ 79%]
[...]
tests/utils/test_server.py ..                                                                                                                                                     [100%]

======================================================================================= FAILURES ========================================================================================
_______________________________________________________________________________ test_snitun_runner_updown _______________________________________________________________________________

    async def test_snitun_runner_updown():
        """Test SniTun Server runner object."""
        server = SniTunServer(
            FERNET_TOKENS, peer_host="127.0.0.1", sni_host="127.0.0.1", sni_port=32000
        )
    
>       await server.start()

tests/server/test_run.py:29: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

fs = {<coroutine object PeerListener.start at 0x7fcddc0b2180>, <coroutine object SNIProxy.start at 0x7fcddc0b0ba0>}

    async def wait(fs, *, timeout=None, return_when=ALL_COMPLETED):
        """Wait for the Futures or Tasks given by fs to complete.
    
        The fs iterable must not be empty.
    
        Coroutines will be wrapped in Tasks.
    
        Returns two sets of Future: (done, pending).
    
        Usage:
    
            done, pending = await asyncio.wait(fs)
    
        Note: This does not raise TimeoutError! Futures that aren't done
        when the timeout occurs are returned in the second set.
        """
        if futures.isfuture(fs) or coroutines.iscoroutine(fs):
            raise TypeError(f"expect a list of futures, not {type(fs).__name__}")
        if not fs:
            raise ValueError('Set of Tasks/Futures is empty.')
        if return_when not in (FIRST_COMPLETED, FIRST_EXCEPTION, ALL_COMPLETED):
            raise ValueError(f'Invalid return_when value: {return_when}')
    
        fs = set(fs)
    
        if any(coroutines.iscoroutine(f) for f in fs):
>           raise TypeError("Passing coroutines is forbidden, use tasks explicitly.")
E           TypeError: Passing coroutines is forbidden, use tasks explicitly.

/usr/lib64/python3.11/asyncio/tasks.py:415: TypeError
--------------------------------------------------------------------------------- Captured stderr setup ---------------------------------------------------------------------------------
DEBUG:asyncio:Using selector: EpollSelector
---------------------------------------------------------------------------------- Captured log setup -----------------------------------------------------------------------------------
DEBUG    asyncio:selector_events.py:54 Using selector: EpollSelector
------------------------------------------------------------------------------- Captured stderr teardown --------------------------------------------------------------------------------
DEBUG:asyncio:Using selector: EpollSelector
--------------------------------------------------------------------------------- Captured log teardown ---------------------------------------------------------------------------------
DEBUG    asyncio:selector_events.py:54 Using selector: EpollSelector
================================================================================ short test summary info ================================================================================
FAILED tests/server/test_run.py::test_snitun_runner_updown - TypeError: Passing coroutines is forbidden, use tasks explicitly.
============================================================================ 1 failed, 103 passed in 28.68s ===========================================================================
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -14,8 +14,10 @@ jobs:
     strategy:
       matrix:
         python-version:
-          - 3.8
-          - 3.9
+          - "3.8"
+          - "3.9"
+          - "3.10"
+          - "3.11"
     steps:
       - uses: actions/checkout@v2
       - name: Set up Python 3.8

--- a/snitun/client/client_peer.py
+++ b/snitun/client/client_peer.py
@@ -62,14 +62,14 @@ class ClientPeer:
                 reader, writer = await asyncio.open_connection(
                     host=self._snitun_host, port=self._snitun_port
                 )
-        except OSError as err:
-            raise SniTunConnectionError(
-                f"Can't connect to SniTun server {self._snitun_host}:{self._snitun_port} with: {err}"
-            ) from err
         except asyncio.TimeoutError:
             raise SniTunConnectionError(
                 f"Connection timeout for SniTun server {self._snitun_host}:{self._snitun_port}",
             ) from None
+        except OSError as err:
+            raise SniTunConnectionError(
+                f"Can't connect to SniTun server {self._snitun_host}:{self._snitun_port} with: {err}"
+            ) from err
 
         # Send fernet token
         writer.write(fernet_token)
@@ -90,6 +90,10 @@ class ClientPeer:
 
                 writer.write(crypto.encrypt(answer))
                 await writer.drain()
+        except asyncio.TimeoutError:
+            raise SniTunConnectionError(
+                "Challenge/Response timeout error to SniTun server"
+            ) from None
         except (
             MultiplexerTransportDecrypt,
             asyncio.IncompleteReadError,
@@ -98,10 +102,6 @@ class ClientPeer:
             raise SniTunConnectionError(
                 f"Challenge/Response error with SniTun server ({err})"
             ) from err
-        except asyncio.TimeoutError:
-            raise SniTunConnectionError(
-                "Challenge/Response timeout error to SniTun server"
-            ) from None
 
         # Run multiplexer
         self._multiplexer = Multiplexer(

--- a/snitun/server/listener_sni.py
+++ b/snitun/server/listener_sni.py
@@ -155,13 +155,13 @@ class SNIProxy:
                     # Flush buffer
                     await writer.drain()
 
-        except (MultiplexerTransportError, OSError, RuntimeError, ConnectionResetError):
-            _LOGGER.debug("Transport closed by Proxy for %s", channel.id)
+        except asyncio.TimeoutError:
+            _LOGGER.debug("Close TCP session after timeout for %s", channel.id)
             with suppress(MultiplexerTransportError):
                 await multiplexer.delete_channel(channel)
 
-        except asyncio.TimeoutError:
-            _LOGGER.debug("Close TCP session after timeout for %s", channel.id)
+        except (MultiplexerTransportError, OSError, RuntimeError, ConnectionResetError):
+            _LOGGER.debug("Transport closed by Proxy for %s", channel.id)
             with suppress(MultiplexerTransportError):
                 await multiplexer.delete_channel(channel)
 

--- a/snitun/server/run.py
+++ b/snitun/server/run.py
@@ -53,14 +53,24 @@ class SniTunServer:
 
         Return coroutine.
         """
-        return asyncio.wait([self._list_peer.start(), self._list_sni.start()])
+        return asyncio.wait(
+            [
+                asyncio.create_task(self._list_peer.start()),
+                asyncio.create_task(self._list_sni.start()),
+            ]
+        )
 
     def stop(self) -> Awaitable[None]:
         """Stop server.
 
         Return coroutine.
         """
-        return asyncio.wait([self._list_peer.stop(), self._list_sni.stop()])
+        return asyncio.wait(
+            [
+                asyncio.create_task(self._list_peer.stop()),
+                asyncio.create_task(self._list_sni.stop()),
+            ]
+        )
 
 
 class SniTunServerSingle:

